### ISSUE TITLE
Failure to create AKS cluster

### ISSUE THREAD

**Issue by qbert** [rancher/rancher#19186]
After creating a SP following the linked instructions ("You can find instructions on how to create an Azure AD Service Principal here.") the command fails with

_failed to get orchestrators: containerservice.ContainerServicesClient#ListOrchestrators: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthorizationFailed" Message="The client 'feb541c9-7f09-4fae-9838-9d8e97debd83' with object id 'feb541c9-7f09-4fae-9838-9d8e97debd83' does not have authorization to perform action 'Microsoft.ContainerService/locations/orchestrators/read' over scope '/subscriptions/2c06ae4f-8140-4af1-9dbe-d69dafb3aa59'._

strangely enough the logged client ID is not what I provided in the form (which was _15132ddd-566f-48fb-9bc2-de33342328a9_).

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v2.2.0` UI: `v2.2.19` |
|Route|`global-admin.clusters.new`|

**Comment by cjellick** [comment]
@qbert did you upgrade from 2.1.x or is this a fresh install of Rancher?

**Comment by cjellick** [comment]
Did this work for you before? Have you tried it in previous rancher versions?

**Comment by qbert** [comment]
@cjellick 
* fresh install of 2.2.2
* so no it didn't work on this before
* i tried creating on azure and importing. That worked

**Comment by jiaqiluo** [comment]
Hi @qbert  I cannot reproduce the error you saw in Rancher v2.2.1
The error message says the client does not have the authorization to perform action 'Microsoft.ContainerService/locations/orchestrators/read', which sounds like an issue of the account's permission. Could you check your account's permission to make sure you have the permission to provision AKS cluster on that specific location?

**Comment by qbert** [comment]
@jiaqiluo 
Thats what I did by provisionig AKS in Azure and importing into Rancher

Pls read the description. The error reads 'client 'feb541c9-7f09-4fae-9838-9d8e97debd83''. That is not my client ID. Looks to me like a wrong client ID is used

**Comment by jiaqiluo** [comment]
@qbert  Correct me if I am wrong, you said that 
- you cannot add an AKS cluster directly from Rancher UI because of the error message you shared in the issue, and even the client ID is wrong in the error message 
- but you can import an existing AKS into Rancher with no problem

Is it right?

**Comment by qbert** [comment]
@jiaqiluo 
Yes that are the two things I tried. But I can't CREATE an AKS cluster from the UI

**Comment by alena1108** [comment]
@qbert have you ever tried provisioning AKS cluster on a previous versions - 2.0.x, 2.1.x? Or are you trying it for the first time on Rancher 2.2.2

**Comment by qbert** [comment]
@alena1108 
Not on a 2.x.x vesion. I tried Rancher on a 1.x version.

**Comment by deniseschannon** [comment]
@qbert In 1.x, you must have been using azure nodes to provision a Rancher launched k8s cluster versus using AKS (where Azure hosts the k8s). They have different requirements. 

As @jiaqiluo has mentioned it sounds as if your permissions that is passed to Rancher for cluster creation has some issues. 

Importing clusters or even using Azure nodes to get a Rancher launched k8s cluster is very different than AKS.

**Comment by qbert** [comment]
@deniseschannon 
But that can't be the issue. It's a clean install of Rancher 2.x, and the error message from Azure is a completely unrelated user_id. The stuff I used with Rancher 1.x is long gone. I tried to provision a new AKS cluster and Rancher apparently used a wrong user_id. Which is a bit hard to figure out because Rancher doesn't show the progress of the provisioning and it's unclear what I'm actually seeing in the error message.

Your telling me that I'm the only one unable to provision an AKS cluster?

**Comment by deniseschannon** [comment]
AKS Clusters are [hoasted k8s providers](https://rancher.com/docs/rancher/v2.x/en/cluster-provisioning/hosted-kubernetes-clusters/) where the hosted provider manages all the k8s components and the credentials are different from using [nodes from azure](https://rancher.com/docs/rancher/v2.x/en/cluster-provisioning/rke-clusters/node-pools/) to create a [Rancher launched k8s cluster](https://rancher.com/docs/rancher/v2.x/en/cluster-provisioning/rke-clusters/). In the latter case, which is what was done in Rancher 1.x, is where Rancher launches and manages the k8s cluster.

I'm just indicating that you can't compare AKS to whatever was done in 1.x. If you wanted a similar experience, you'd use the azure nodes to have rancher deploy.

In every release, we've tested deploying AKS clusters and there are known issues with AKS flakiness that have been attempted to be fixed in 2.2.

I haven't seen anyone hitting your issue where credentials are not working
